modules/src/events.rs

```diff
@@ -322,15 +333,18 @@ impl TryFrom<IbcEvent> for AbciEvent {
 // This is tendermint specific
 pub fn from_tx_response_event(height: Height, event: &tendermint::abci::Event) -> Option<IbcEvent> {
     // Return the first hit we find
-    if let Some(mut client_res) = ClientEvents::try_from_tx(event) {
+    if let Some(mut client_res) = client_events::try_from_tx(event) {
         client_res.set_height(height);
         Some(client_res)
-    } else if let Some(mut conn_res) = ConnectionEvents::try_from_tx(event) {
+    } else if let Some(mut conn_res) = connection_events::try_from_tx(event) {
         conn_res.set_height(height);
         Some(conn_res)
-    } else if let Some(mut chan_res) = ChannelEvents::try_from_tx(event) {
+    } else if let Some(mut chan_res) = channel_events::try_from_tx(event) {
         chan_res.set_height(height);
         Some(chan_res)
+    } else if let Some(mut gov_res) = gov_events::try_from_tx(event) {
+        gov_res.set_height(height);
```

Comment:
I have cargo-culted this from the other event types; not sure if it makes sense in this case.

Reply:
I have a slight preference to leave `ibc::events::from_tx_response_event` untouched, to remain an IBC-module-specific method, with IBC specific functionality.

Cosmos SDK-specific stuff can live in the `ibc-relayer` crate, since only the relayer needs it (not the IBC module itself). So the suggestion would be to:

- undo the changes in `from_tx_response_event`
- move `modules/src/cosmos/gov/events.rs` -> `relayer/src/chain/cosmos/events.rs`
- modify method [all_ibc_events_from_tx_search_response](https://github.com/informalsystems/ibc-rs/blob/b3f9e4ce39cd3133ae9d88de1541f9f795e9ddc9/relayer/src/chain/cosmos.rs#L2229) to handle two types of events:
    - ibc events, which [it does already](https://github.com/informalsystems/ibc-rs/blob/b3f9e4ce39cd3133ae9d88de1541f9f795e9ddc9/relayer/src/chain/cosmos.rs#L2244-L2247) via method `from_tx_response_event`
    - sdk events, which it can do via the new module `cosmos::events::gov_events::try_from_tx(..)`
- we should also rename 

The solution could look as follows:

```diff
diff --git a/relayer/src/chain/cosmos.rs b/relayer/src/chain/cosmos.rs
index e07f0a5d..f817b167 100644
--- a/relayer/src/chain/cosmos.rs
+++ b/relayer/src/chain/cosmos.rs
@@ -2226,7 +2226,7 @@ fn update_client_from_tx_search_response(
         .map(IbcEvent::UpdateClient)
 }

-fn all_ibc_events_from_tx_search_response(chain_id: &ChainId, response: ResultTx) -> Vec<IbcEvent> {
+fn all_events_from_tx_search_response(chain_id: &ChainId, response: ResultTx) -> Vec<IbcEvent> {
     let height = ICSHeight::new(chain_id.version(), u64::from(response.height));
     let deliver_tx_result = response.tx_result;
     if deliver_tx_result.code.is_err() {
@@ -2245,6 +2245,10 @@ fn all_ibc_events_from_tx_search_response(chain_id: &ChainId, response: ResultTx
             trace!("got IBC event: {:#?}", ibc_ev);
             result.push(ibc_ev);
         }
+        if let Some(sdk_ev) = cosmos::events::sdk_events::from_tx_response_event(height, &event) {
+            trace!("got SDK event: {:#?}", ibc_ev);
+            result.push(sdk_ev);
+        }
     }
```

Reply:
The difficulties I've run into with the recommended approach is, the result of `ChainHandle::query_txs` et al. is currently `Vec<IbcEvent>`, and remaking all of these uses to return a new combined event type is quite a slog. @adizere do you think we should commit to such a redesign, or could there be another way to solve this?

I've tried to come up with a simpler solution, special-casing the SDK event handling aside from the generic `ChainHandle` API and adding a cosmos-specific method to call directly, but such a refactor looks hackish and I feel that we should not do this for every code path where an SDK-specific event might be used to confirm a transaction.